ucubot/Controllers/LessonSignalEndpointController.cs

```diff
@@ -41,15 +98,29 @@ public LessonSignalDto ShowSignal(long id)
             var userId = message.user_id;
             var signalType = message.text.ConvertSlackMessageToSignalType();
 
-            // TODO: add insert command to store signal
+            var connectionString = _configuration.GetConnectionString("BotDatabase");
+            var conn = new MySqlConnection(connectionString);
+            conn.Open();
+            
+            var command = new MySqlCommand(String.Format("INSERT INTO lesson_signal (user_id, signal_type) VALUES ({0}, {1});", userId, signalType), conn);
```

Comment:
fail: Microsoft.AspNetCore.Server.Kestrel[13]
app_1   |       Connection id "0HLC53GE3VE9H", Request id "0HLC53GE3VE9H:00000002": An unhandled exception was thrown by the application.
app_1   | MySql.Data.MySqlClient.MySqlException (0x80004005): Unknown column 'U111' in 'field list'
app_1   |    at MySql.Data.MySqlClient.MySqlStream.ReadPacket()
app_1   |    at MySql.Data.MySqlClient.NativeDriver.GetResult(Int32& affectedRow, Int64& insertedId)
app_1   |    at MySql.Data.MySqlClient.Driver.NextResult(Int32 statementId, Boolean force)
app_1   |    at MySql.Data.MySqlClient.MySqlDataReader.NextResult()
app_1   |    at MySql.Data.MySqlClient.MySqlCommand.ExecuteReader(CommandBehavior behavior)
app_1   |    at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQuery()
app_1   |    at System.Data.Common.DbCommand.ExecuteNonQueryAsync(CancellationToken cancellationToken)
app_1   | --- End of stack trace from previous location where exception was thrown ---
app_1   |    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
app_1   |    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
app_1   |    at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
app_1   |    at ucubot.Controllers.LessonSignalEndpointController.<CreateSignal>d__4.MoveNext() in /app/Controllers/LessonSignalEndpointController.cs:line 106
app_1   | --- End of stack trace from previous location where exception was thrown ---
app_1   |    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
app_1   |    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
app_1   |    at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
app_1   |    at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.<InvokeActionMethodAsync>d__12.MoveNext()
app_1   | --- End of stack trace from previous location where exception was thrown ---
app_1   |    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
app_1   |    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
app_1   |    at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.<InvokeNextActionFilterAsync>d__10.MoveNext()
app_1   | --- End of stack trace from previous location where exception was thrown ---
app_1   |    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
app_1   |    at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.Rethrow(ActionExecutedContext context)
app_1   |    at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
app_1   |    at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.<InvokeInnerFilterAsync>d__14.MoveNext()
app_1   | --- End of stack trace from previous location where exception was thrown ---
app_1   |    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
app_1   |    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
app_1   |    at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.<InvokeNextResourceFilter>d__22.MoveNext()
app_1   | --- End of stack trace from previous location where exception was thrown ---
app_1   |    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
app_1   |    at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.Rethrow(ResourceExecutedContext context)
app_1   |    at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
app_1   |    at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.<InvokeFilterPipelineAsync>d__17.MoveNext()
app_1   | --- End of stack trace from previous location where exception was thrown ---
app_1   |    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
app_1   |    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
app_1   |    at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.<InvokeAsync>d__15.MoveNext()
app_1   | --- End of stack trace from previous location where exception was thrown ---
app_1   |    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
app_1   |    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
app_1   |    at Microsoft.AspNetCore.Builder.RouterMiddleware.<Invoke>d__4.MoveNext()
app_1   | --- End of stack trace from previous location where exception was thrown ---
app_1   |    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
app_1   |    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
app_1   |    at ucubot.Infrastructure.AspNetCoreExtensions.RequestScopingStartupFilter.<<ConfigureRequestScoping>b__3_0>d.MoveNext() in /app/Infrastructure/AspNetCoreExtensions.cs:line 94
app_1   | --- End of stack trace from previous location where exception was thrown ---
app_1   |    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
app_1   |    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
app_1   |    at Microsoft.AspNetCore.Hosting.Internal.RequestServicesContainerMiddleware.<Invoke>d__3.MoveNext()
app_1   | --- End of stack trace from previous location where exception was thrown ---
app_1   |    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
app_1   |    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
app_1   |    at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.Frame`1.<ProcessRequestsAsync>d__2.MoveNext()
app_1   | fail: Microsoft.AspNetCore.Server.Kestrel[13]
app_1   |       Connection id "0HLC53GE3VE9I", Request id "0HLC53GE3VE9I:00000002": An unhandled exception was thrown by the application.
app_1   | MySql.Data.MySqlClient.MySqlException (0x80004005): Unknown column 'U111' in 'field list'
app_1   |    at MySql.Data.MySqlClient.MySqlStream.ReadPacket()
app_1   |    at MySql.Data.MySqlClient.NativeDriver.GetResult(Int32& affectedRow, Int64& insertedId)
app_1   |    at MySql.Data.MySqlClient.Driver.NextResult(Int32 statementId, Boolean force)
app_1   |    at MySql.Data.MySqlClient.MySqlDataReader.NextResult()
app_1   |    at MySql.Data.MySqlClient.MySqlCommand.ExecuteReader(CommandBehavior behavior)
app_1   |    at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQuery()
app_1   |    at System.Data.Common.DbCommand.ExecuteNonQueryAsync(CancellationToken cancellationToken)
app_1   | --- End of stack trace from previous location where exception was thrown ---
app_1   |    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
app_1   |    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
app_1   |    at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
app_1   |    at ucubot.Controllers.LessonSignalEndpointController.<CreateSignal>d__4.MoveNext() in /app/Controllers/LessonSignalEndpointController.cs:line 106
app_1   | --- End of stack trace from previous location where exception was thrown ---
app_1   |    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
app_1   |    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
app_1   |    at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
app_1   |    at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.<InvokeActionMethodAsync>d__12.MoveNext()
app_1   | --- End of stack trace from previous location where exception was thrown ---
app_1   |    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
app_1   |    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
app_1   |    at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.<InvokeNextActionFilterAsync>d__10.MoveNext()
app_1   | --- End of stack trace from previous location where exception was thrown ---
app_1   |    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
app_1   |    at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.Rethrow(ActionExecutedContext context)
app_1   |    at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
app_1   |    at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.<InvokeInnerFilterAsync>d__14.MoveNext()
app_1   | --- End of stack trace from previous location where exception was thrown ---
app_1   |    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
app_1   |    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
app_1   |    at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.<InvokeNextResourceFilter>d__22.MoveNext()
app_1   | --- End of stack trace from previous location where exception was thrown ---
app_1   |    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
app_1   |    at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.Rethrow(ResourceExecutedContext context)
app_1   |    at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
app_1   |    at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.<InvokeFilterPipelineAsync>d__17.MoveNext()
app_1   | --- End of stack trace from previous location where exception was thrown ---
app_1   |    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
app_1   |    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
app_1   |    at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.<InvokeAsync>d__15.MoveNext()
app_1   | --- End of stack trace from previous location where exception was thrown ---
app_1   |    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
app_1   |    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
app_1   |    at Microsoft.AspNetCore.Builder.RouterMiddleware.<Invoke>d__4.MoveNext()
app_1   | --- End of stack trace from previous location where exception was thrown ---
app_1   |    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
app_1   |    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
app_1   |    at ucubot.Infrastructure.AspNetCoreExtensions.RequestScopingStartupFilter.<<ConfigureRequestScoping>b__3_0>d.MoveNext() in /app/Infrastructure/AspNetCoreExtensions.cs:line 94
app_1   | --- End of stack trace from previous location where exception was thrown ---
app_1   |    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
app_1   |    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
app_1   |    at Microsoft.AspNetCore.Hosting.Internal.RequestServicesContainerMiddleware.<Invoke>d__3.MoveNext()
app_1   | --- End of stack trace from previous location where exception was thrown ---
app_1   |    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
app_1   |    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
app_1   |    at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.Frame`1.<ProcessRequestsAsync>d__2.MoveNext()
test_1  | NUnit Adapter 3.9.0.0: Test execution complete
test_1  | Failed   TestGetCreateGetDeleteGet
test_1  | Error Message:
test_1  |  Expected object to be Accepted, but found InternalServerError.
test_1  | Stack Trace:
test_1  |    at FluentAssertions.Execution.LateBoundTestFramework.Throw(String message)
test_1  |    at FluentAssertions.Execution.AssertionScope.FailWith(String message, Object[] args)
test_1  |    at FluentAssertions.Primitives.ObjectAssertions.Be(Object expected, String because, Object[] becauseArgs)
test_1  |    at usubot.End2EndTests.Assignment1Tests.<TestGetCreateGetDeleteGet>d__8.MoveNext() in /app/usubot.End2EndTests/Assignment1Tests.cs:line 126
test_1  | --- End of stack trace from previous location where exception was thrown ---
test_1  |    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
test_1  |    at NUnit.Framework.Internal.AsyncInvocationRegion.AsyncTaskInvocationRegion.WaitForPendingOperationsToComplete(Object invocationResult)
test_1  |    at NUnit.Framework.Internal.Commands.TestMethodCommand.RunAsyncTestMethod(TestExecutionContext context)
test_1  | 
test_1  | Failed   TestSqlInjectionFail
test_1  | Error Message:
test_1  |  Expected object to be Accepted, but found InternalServerError.
test_1  | Stack Trace:
test_1  |    at FluentAssertions.Execution.LateBoundTestFramework.Throw(String message)
test_1  |    at FluentAssertions.Execution.AssertionScope.FailWith(String message, Object[] args)
test_1  |    at FluentAssertions.Primitives.ObjectAssertions.Be(Object expected, String because, Object[] becauseArgs)
test_1  |    at usubot.End2EndTests.Assignment1Tests.<TestSqlInjectionFail>d__9.MoveNext() in /app/usubot.End2EndTests/Assignment1Tests.cs:line 160
test_1  | --- End of stack trace from previous location where exception was thrown ---
test_1  |    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
test_1  |    at NUnit.Framework.Internal.AsyncInvocationRegion.AsyncTaskInvocationRegion.WaitForPendingOperationsToComplete(Object invocationResult)
test_1  |    at NUnit.Framework.Internal.Commands.TestMethodCommand.RunAsyncTestMethod(TestExecutionContext context)
test_1  | 
test_1  | Failed   TestNonExistRecordReturns404
test_1  | Error Message:
test_1  |  System.InvalidOperationException : Sequence contains no elements
test_1  | Stack Trace:
test_1  |    at System.Linq.Enumerable.Max(IEnumerable`1 source)
test_1  |    at usubot.End2EndTests.Assignment1Tests.<TestNonExistRecordReturns404>d__10.MoveNext() in /app/usubot.End2EndTests/Assignment1Tests.cs:line 183
test_1  | --- End of stack trace from previous location where exception was thrown ---
test_1  |    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
test_1  |    at NUnit.Framework.Internal.AsyncInvocationRegion.AsyncTaskInvocationRegion.WaitForPendingOperationsToComplete(Object invocationResult)
test_1  |    at NUnit.Framework.Internal.Commands.TestMethodCommand.RunAsyncTestMethod(TestExecutionContext context)
test_1  | 
test_1  | Total tests: 7. Passed: 4. Failed: 3. Skipped: 0.
test_1  | Test Run Failed.
test_1  | Test execution time: 12.9535 Seconds
romanvey_test_1 exited with code 1